capstone/src/test/java/com/google/sps/LessonDeserializationTest.java

```diff
@@ -0,0 +1,219 @@
+import java.io.IOException;
+import com.google.appengine.api.datastore.Entity;
+
+@RunWith(JUnit4.class)
+public final class LessonDeserializationTest {
+
+    //Helper to set local datastore service
+    private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig()
+          .setDefaultHighRepJobPolicyUnappliedJobPercentage(100));
+
+    private static final String TEST_TITLE = "Piano Lessons";
+    private static final String TEST_DESCRIPTION = "Easy lessons for beginners";
+    private static final String TEST_URL = "www.piano.com";
+    private static final String TEST_EDIT_URL = "www.piano.com/edit";
+
+    @Before
+    public void setUp() {
+        helper.setUp();
+    }
+
+    @After
+    public void tearDown() {
+        helper.tearDown();
+    }
+
+    @Test
+    public void deserializeJsonGeneralTest() {
+        JsonObject jsonObject = new JsonObject();
+        jsonObject.addProperty(Lesson.TYPE_PROPERTY_KEY, Lesson.TYPE_FORM);
+        jsonObject.addProperty(Lesson.TITLE_PROPERTY_KEY, LessonDeserializationTest.TEST_TITLE);
+        jsonObject.addProperty(Lesson.DESCRIPTION_PROPERTY_KEY, LessonDeserializationTest.TEST_DESCRIPTION);
+        jsonObject.addProperty(Form.EDIT_URL_PROPERTY_KEY, LessonDeserializationTest.TEST_URL);
+        jsonObject.addProperty(Form.URL_PROPERTY_KEY, LessonDeserializationTest.TEST_EDIT_URL);
+
+        try{ 
+            Lesson lesson = Lesson.deserializeJson(jsonObject.toString());
+
+            Assert.assertEquals(lesson.getType(), Lesson.TYPE_FORM);
+            Assert.assertEquals(lesson.getTitle(), LessonDeserializationTest.TEST_TITLE);
+            Assert.assertEquals(lesson.getDescription(), LessonDeserializationTest.TEST_DESCRIPTION);
+            Assert.assertEquals(lesson.getIsDraft(), false);
+            Assert.assertTrue(lesson instanceof Form);
+            Form form = (Form) lesson; 
+            Assert.assertEquals(form.getEditUrl(), LessonDeserializationTest.TEST_URL);
+            Assert.assertEquals(form.getUrl(), LessonDeserializationTest.TEST_EDIT_URL);
+            
+        } catch (IOException e) {
+            Assert.fail("IOException Thrown");
+        }
+    }
+
+    @Test
+    public void deserializeJsonMissingFieldTest() {
+        JsonObject jsonObject = new JsonObject();
+        jsonObject.addProperty(Lesson.TYPE_PROPERTY_KEY, Lesson.TYPE_FORM);
+        jsonObject.addProperty(Lesson.DESCRIPTION_PROPERTY_KEY, LessonDeserializationTest.TEST_DESCRIPTION);
+        jsonObject.addProperty(Form.EDIT_URL_PROPERTY_KEY, LessonDeserializationTest.TEST_URL);
+        jsonObject.addProperty(Form.URL_PROPERTY_KEY, LessonDeserializationTest.TEST_EDIT_URL);
+
+        try{ 
+            Lesson lesson = Lesson.deserializeJson(jsonObject.toString());
+            Assert.fail();
+        } catch(NullPointerException e) {
+            return;
+        } catch(Exception e) {
+            Assert.fail();
+        }
```

Comment:
This is a test to see deserializeJson fail and throw a NullPointerException (when it should). Was wondering if this is the right way to catch the error we are looking for unless there is another "best practice" way.

Reply:
I see, I threw an IllegalArgumentException instead of IOException.